packages/cli/src/credentials/credentials.controller.ee.ts

```diff
@@ -190,15 +190,19 @@ EECredentialsController.put(
 			sharees_removed: amountRemoved,
 		});
 
+		const isEmailingEnabled = config.getEnv('userManagement.emails.mode') === 'smtp';
+
+		if (!isEmailingEnabled) return;
+
 		const recipients = await Container.get(UserRepository).getEmailsByIds(newShareeIds);
 
 		if (recipients.length === 0) return;
 
 		try {
 			await Container.get(UserManagementMailer).notifyCredentialsShared({
-				sharerFirstName: req.user.firstName,
+				sharer: { id: req.user.id, firstName: req.user.firstName },
+				newShareeIds,
 				credentialsName: credential.name,
-				recipientEmails: recipients.map(({ email }) => email),
 				baseUrl: Container.get(UrlService).getInstanceBaseUrl(),
```

Comment:
passing around `baseUrl` from function to function is legacy code. we should update `UserManagementMailer` to use `UrlService` directly.